torch/nn/parallel/distributed.py

```diff
@@ -305,14 +305,19 @@ class DistributedDataParallel(Module):
 
     Args:
         module (Module): module to be parallelized
-        device_ids (list of int or torch.device): CUDA devices. This should
-                   only be provided when the input module resides on a single
-                   CUDA device. For single-device modules, the i'th
-                   :attr:`module` replica is placed on ``device_ids[i]``. For
-                   multi-device modules and CPU modules, ``device_ids`` must be
-                   ``None`` or an empty list, and input data for the forward
-                   pass must be placed on the correct device. (default: all
-                   visible devices for single-device modules)
+        device_ids (list of int or torch.device): CUDA devices.
+                   1) For single-device modules, ``device_ids`` should only
+                   contain exactly one device id, which represents the only
+                   CUDA device where the input module resides.
+                   Alternatively, if ``None`` is provided, the current device
+                   will be added to ``device_ids``. This requires calling
+                   ``torch.cuda.set_device`` to set the appropriate device
+                   before calling DDP constructor.
```

Comment:
Lets go with Yanli's suggestion where if `None` is provided, we expect the user to place the model on the correct device instead of expecting `set_device` (this is consistent with multi-device models).

Also, avoid adding statements like 'the current device will be added to ``device_ids``', since that is implementation detail that the user shouldn't worry about.